constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/LinearEquation.java

```diff
@@ -0,0 +1,180 @@
+    );
+  }
+
+  public boolean changing() {
+    return this.rate != 0;
+  }
+
+  private Optional<Duration> intersectionPointWith(final LinearEquation other) {
+    if (this.rate == other.rate) return Optional.empty();
+    return Optional.of(
+        this.initialTime.plus(
+            Duration.roundNearest(
+                (other.valueAt(this.initialTime) - this.initialValue) / (this.rate - other.rate)*1000000,
+                Duration.MICROSECONDS)
+        )
+    );
+  }
+
+  public Windows intervalsLessThan(final LinearEquation other) {
+    return getInequalityIntervals(other, (l, r) -> l < r);
+  }
+
+  public Windows intervalsLessThanOrEqualTo(final LinearEquation other) {
+    return getInequalityIntervals(other, (l, r) -> l <= r);
+  }
+
+  public Windows intervalsGreaterThan(final LinearEquation other) {
+    return getInequalityIntervals(other, (l, r) -> l > r);
+  }
+
+  public Windows intervalsGreaterThanOrEqualTo(final LinearEquation other) {
+    return getInequalityIntervals(other, (l, r) -> l >= r);
+  }
+
+  private Windows getInequalityIntervals(
+      final LinearEquation other,
+      final BiFunction<Double, Double, Boolean> op
+  ) {
+    final var intersectionOption = this.intersectionPointWith(other);
+
+    if (intersectionOption.isEmpty()) {
+      final var resultEverywhere = op.apply(this.initialValue, other.valueAt(this.initialTime));
+      return new Windows(resultEverywhere);
+    } else {
+      final var intersection = intersectionOption.get();
+      final var builder = IntervalMap.<Boolean>builder();
+      final var oneSecondBefore = intersection.minus(Duration.SECOND);
+      final var oneSecondAfter = intersection.plus(Duration.SECOND);
+      builder.set(
+          Interval.between(Duration.MIN_VALUE, Inclusive, intersection, Exclusive),
+          op.apply(this.valueAt(oneSecondBefore), other.valueAt(oneSecondBefore))
+      );
+      builder.set(
+          Interval.at(intersection),
+          op.apply(this.valueAt(intersection), other.valueAt(intersection))
+      );
+      builder.set(
+          Interval.between(intersection, Exclusive, Duration.MAX_VALUE, Inclusive),
+          op.apply(this.valueAt(oneSecondAfter), other.valueAt(oneSecondAfter))
+      );
+      return new Windows(builder.build());
+    }
+  }
+
+  public Windows intervalsEqualTo(final LinearEquation other) {
+    final var intersection = this.intersectionPointWith(other);
+
+    if (intersection.isEmpty()) {
+      return new Windows(this.initialValue == other.valueAt(this.initialTime));
+    } else {
+      final var t = intersection.get();
+      final var builder = IntervalMap.<Boolean>builder();
+      builder.set(
+          Interval.between(Duration.MIN_VALUE, Inclusive, t, Exclusive),
+          false
+      );
+      builder.set(Interval.at(t), true);
+      builder.set(
+          Interval.between(t, Exclusive, Duration.MAX_VALUE, Inclusive),
+          false
+      );
+      return new Windows(builder.build());
```

Comment:
Consider replacing the builder here with the Windows varargs constructor:

```suggestion
      return new Windows(
          Segment.of(Interval.between(Duration.MIN_VALUE, Inclusive, t, Exclusive), false),
          Segment.of(Interval.at(t), true),
          Segment.of(Interval.between(t, Exclusive, Duration.MAX_VALUE, Inclusive), false));
```